packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.exception.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  ConflictError,
+  ForbiddenError,
+  InternalServerError,
+  NotFoundError,
+  UserInputError,
+} from 'src/engine/utils/graphql-errors.util';
+import { CustomError } from 'src/utils/custom-error';
+
+export class FieldMetadataException extends CustomError {
+  code: FieldMetadataExceptionCode;
+  constructor(message: string, code: FieldMetadataExceptionCode) {
+    super(message, code);
+  }
+}
+
+export enum FieldMetadataExceptionCode {
+  FIELD_METADATA_NOT_FOUND = 'FIELD_METADATA_NOT_FOUND',
+  INVALID_FIELD_INPUT = 'INVALID_FIELD_INPUT',
+  FIELD_MUTATION_NOT_ALLOWED = 'FIELD_MUTATION_NOT_ALLOWED',
+  FIELD_ALREADY_EXISTS = 'FIELD_ALREADY_EXISTS',
+  OBJECT_METADATA_NOT_FOUND = 'OBJECT_METADATA_NOT_FOUND',
+  INTERNAL_SERVER_ERROR = 'INTERNAL_SERVER_ERROR',
+}
+
+export const fieldMetadataExceptionHandler = (error: Error) => {
```

Comment:
I'm not sure about this. It seems wrong to me that the exception itself contains logic on how to translate to GraphQL error. I think it could depend of the context.

Ex: for a given graphql resolver, having a OBJECT_METADATA_NOT_FOUND is a 404 error (this can happen and we warn the user) but for another one it could be a 500 (this should never happen)